modules/modal/react/lib/ModalContent.tsx

```diff
@@ -0,0 +1,178 @@
+   * If this ref is not provided the modal will try to use the close icon. If that icon is not available,
+   * it will make the modal heading focusable and focus on that instead.
+   */
+  firstFocusRef?: React.RefObject<HTMLElement>;
+}
+
+const fadeIn = keyframes`
+  from {
+    background: none;
+  }
+  to {
+    background: rgba(0,0,0,0.65);
+  }
+`;
+
+const Container = styled('div')({
+  position: 'fixed',
+  top: 0,
+  left: 0,
+  width: '100vw',
+  height: '100vh',
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  background: 'rgba(0,0,0,0.65)',
+  animationName: `${fadeIn}`,
+  animationDuration: '0.3s',
+  zIndex: 1,
+});
+
+const transformOrigin = {
+  horizontal: 'center',
+  vertical: 'bottom',
+} as const;
+
+function getFirstElementToFocus(modalEl: HTMLElement): HTMLElement {
+  const firstFocusable = modalEl.querySelector<HTMLElement>(
+    `[data-close=close],[id="${modalEl.getAttribute('aria-labelledby')}"]`
+  );
+  if (firstFocusable) {
+    if (firstFocusable.tagName === 'H3') {
+      // If there is no close icon, we need to transfer focus to the header.
+      // Setting tabIndex allows the header to be focusable.
+      // We do the header instead of the next focusable element to prevent useful context from being skipped
+      firstFocusable.tabIndex = 0;
+      firstFocusable.style.outline = 'none';
+
+      const changeTabIndex = () => {
+        // We no longer need to focus on the header after it loses focus
+        // We simply want to transfer focus inside
+        firstFocusable.removeEventListener('blur', changeTabIndex);
+        // We must wait one frame to ensure tabbable checks are satisfied
+        // by all focus libraries...
+        requestAnimationFrame(() => {
+          firstFocusable.removeAttribute('tabIndex');
+        });
+      };
+      firstFocusable.addEventListener('blur', changeTabIndex);
+    }
+    return firstFocusable;
+  } else {
+    throw new Error(
+      'No focusable element was found. Please ensure modal has at least one focusable element'
+    );
+  }
+}
+
+const useKeyDownListener = (handleKeydown: EventListenerOrEventListenerObject) => {
+  React.useEffect(() => {
+    document.addEventListener('keydown', handleKeydown);
+    return () => {
+      document.removeEventListener('keydown', handleKeydown);
+    };
+  }, [handleKeydown]);
+};
+
+const useInitialFocus = (
+  modalRef: React.RefObject<HTMLElement>,
+  firstFocusRef: React.RefObject<HTMLElement> | undefined
+) => {
+  const handlerRef = document.activeElement instanceof HTMLElement ? document.activeElement : null;
+
+  React.useEffect(() => {
+    if (modalRef.current) {
+      const elem =
+        (firstFocusRef && firstFocusRef.current) || getFirstElementToFocus(modalRef.current);
+      elem.focus();
+    }
+    return () => {
+      if (handlerRef) {
+        handlerRef.focus();
+      }
+    };
+  }, [modalRef, firstFocusRef]);
+};
+
+const ModalContent = ({
```

Comment:
The ModalContent was moved out of the `Modal` component to prevent processing if `open` was false.